### ISSUE TITLE
Theia LLM - First iteration

### ISSUE THREAD

**Issue by sdirix** [eclipse-theia/theia#14143]
## Overview

This is a ticket to manage the direct follow up of the initial Theia AI support contribution (#14048)

### Architecture

These tasks / evaluations challenge major parts of the contributed Theia AI architecture

- The initial contribution adds an own AI variable service. Theia already has a generic variable service. [It must be evaluated whether they should be fully merged instead of just integrated as they are right now](https://github.com/eclipse-theia/theia/pull/14048#discussion_r1734322460).
- The AI configuration view partially competes with the preferences. It must be evaluated whether they can be (partially) merged and how the new role of the configuration view then looks like.

### Major impact

- The original contribution customizes the `EditorManager` in the `AIEditorManager` and rebinds it. [This is problematic for downstream projects which also rebind the `EditorManager`](https://github.com/eclipse-theia/theia/pull/14048#discussion_r1734227614). The editor manager enhancements should be merged into the base implementation instead. 
- The agent inheritance tree is quite large already. It must be checked whether it can be simplified, [including using composition patterns over inheritance](https://github.com/eclipse-theia/theia/pull/14048/files#r1736144466). 
- [It must be discussed whether we want to adapt the ai code completion items or remove them](https://github.com/eclipse-theia/theia/pull/14048/files#r1742286819). We will definitively keep the inline suggestions.  
- User edited prompt templates are stored in separate configurable directories. [For this a already layered mechanism can be used](https://github.com/eclipse-theia/theia/pull/14048#discussion_r1741016957).

### Best practices

- Rename and refactor the language model delegation mechanic. Some of the used classes/services can be merged and they should [follow the Theia conventions for naming](https://github.com/eclipse-theia/theia/pull/14048/files#r1740906884).
- Introduce i18n for all hard coded strings
- A user message was shown by customizing the `MonacoEditorProvider` in the same way as the `EditorManager`. This was removed from the contribution. However it would be great to get the customization back in. This must be evaluated.
- The contribution introduces a new offset interface. [It should be checked whether it can be replaced with an already existing concept in Theia](https://github.com/eclipse-theia/theia/pull/14048/files#r1736491182).

### Minor scale improvements

- Prompts implementation [can be improved](https://github.com/eclipse-theia/theia/pull/14048#discussion_r1740966401)
- Some of the new variables are contributed one at a time in separate contributions. [They should be merged](https://github.com/eclipse-theia/theia/pull/14048/files#r1741024476).
- [Improve on agent interface namings](https://github.com/eclipse-theia/theia/pull/14048#discussion_r1741601440)
- Rework the language model implementation ([methods](https://github.com/eclipse-theia/theia/pull/14048#discussion_r1740937483), [jsdoc](https://github.com/eclipse-theia/theia/pull/14048#discussion_r1740938730), etc.)
- [Inline constants which are used only once](https://github.com/eclipse-theia/theia/pull/14048#discussion_r1740940779)
- [Improve AI history implementation](https://github.com/eclipse-theia/theia/pull/14048#discussion_r1741034003)
- [Improve AI terminal implementation](https://github.com/eclipse-theia/theia/pull/14048#discussion_r1741050376)

**Comment by eneufeld** [comment]
@ndoschek @sdirix I suggest to close this one and pull out what is actually still open in follow ups. wdyt?

**Comment by sdirix** [comment]
I can look through the open topics and will create follow ups
